packages/core/src/readers/SimpleDirectoryReader.ts

```diff
@@ -37,10 +35,10 @@ export class TextFileReader implements BaseReader {
 
 export const FILE_EXT_TO_READER: Record<string, BaseReader> = {
```

Comment:
@himself65 

**Suggestion**, we include a minimal `FILE_EXT_TO_READER` as default (without the readers that are not working with edge) and make it easy for the dev to add whatever readers that they need, e.g.:

To use all readers (importing `ALL_READERS` would import the non-edge readers):
```
const documents = await new SimpleDirectoryReader({
    readers: ALL,
  }).loadData({
    directoryPath: "node_modules/llamaindex/examples",
  })
```

or to use specific readers
```
const documents = await new SimpleDirectoryReader({
    readers: {
    "pdf": PDFReader
    },
  }).loadData({
    directoryPath: "node_modules/llamaindex/examples",
  })
```

Reply:
no, there is no directory in edge runtime, you cannot access file system

Reply:
AFAIK you have read access to the file system, but not write access